projects/plugins/boost/app/rest-api/endpoints/Image_Analysis_Fix.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\REST_API\Endpoints;
+
+use Automattic\Jetpack_Boost\Admin\Config;
+use Automattic\Jetpack_Boost\Lib\Premium_Features;
+use Automattic\Jetpack_Boost\Modules\Image_Size_Analysis\Image_Size_Analysis_Fixer;
+use Automattic\Jetpack_Boost\REST_API\Contracts\Endpoint;
+use Automattic\Jetpack_Boost\REST_API\Permissions\Current_User_Admin;
+
+/**
+ * API Endpoint for starting an Image Size Analysis run.
+ * Lives at POST jetpack-boost/v1/image-size-analysis/fix
+ */
+class Image_Analysis_Fix implements Endpoint {
+
+	const NONCE_ACTION = Config::FIX_IMAGE_DIMENSIONS_NONCE;
+
+	public function request_methods() {
+		return \WP_REST_Server::EDITABLE;
+	}
+
+	public function response( $request ) {
+		// @TODO: Add a proper feature flag for this instead of just checking if priority support available.
+		if ( ! Premium_Features::has_feature( Premium_Features::IMAGE_SIZE_ANALYSIS ) ) {
+			return new \WP_Error( 'not-allowed', 'Feature not enabled' );
+		}
+
+		if ( empty( $request->get_param( 'nonce' ) ) || ! wp_verify_nonce( sanitize_key( $request->get_param( 'nonce' ) ), self::NONCE_ACTION ) ) {
+			return new \WP_Error( 'not-allowed', 'Bad Nonce' );
+		}
+
+		$params = $request->get_params();
+		unset( $params['nonce'] );
+
+		$fixes = Image_Size_Analysis_Fixer::get_fixes( $request->get_param( 'post_id' ) );
+
+		$image_url = Image_Size_Analysis_Fixer::fix_url( $request->get_param( 'image_url' ) );
+
+		$attachment_id = attachment_url_to_postid( esc_url( $image_url ) );
+		if ( $attachment_id ) {
+			$fixes[ $attachment_id ] = $params;
+		} else {
+			$fixes[ md5( $image_url ) ] = $params; // hot linked image, possibly from another site.
+		}
+
+		update_post_meta( $request->get_param( 'post_id' ), 'jb_image_fixes', $fixes );
+
+		// Send a success response.
+		return rest_ensure_response(
+			array(
+				'ok' => true,
+			)
+		);
```

Comment:
We're sending `ok` even if `update_post_meta` failed here.

Reply:
I've just addressed that. The button won't toggle to the other state if no changes were made or saved. There's no indication in the UI, apart from the button not changing.